packages/astro/src/core/errors/README.md

```diff
@@ -89,6 +90,19 @@ Example:
 
 The `@message` property is intended to provide slightly more context when it is helpful: a more descriptive error message or a collection of common messages if there are multiple possible error messages. Try to avoid making substantial changes to existing messages so that they are easy to find for users who copy and search the exact content of an error message.
 
+### Removing errors
+
+If an error is no longer relevant, it can deprecated by adding a `@deprecated` tag to the JSDoc comment with a message that will be shown in the docs. This is useful for users on previous versions who might still encounter the error to know that perhaps upgrading to a newer version of Astro would solve their issue.
```

Comment:
IIUC, an error is no longer relevant and marked deprecated if it's still being used by the codebase, but in a deprecated logic path?

If we no longer use the error, would we completely remove it?

Reply:
I see! I think I'm confused in deciding when would a deprecation message be required or not required then. For me if I removed a feature that no longer uses an error, I could remove the error directly. And if I tweak how a feature works that the error is no longer relevant, I'd use a deprecation message?

I think some examples would be nice, but not a big deal if not. Was worried if someone could misjudge when to do which.